spec/models/device_spec.rb

```diff
@@ -31,9 +31,12 @@ def should_be_valid_with_instition_and_site(device)
 
   context "validations" do
 
-    it { is_expected.to validate_presence_of :device_model }
-    it { is_expected.to validate_presence_of :name }
-    it { is_expected.to validate_presence_of :institution }
+    it do
+      expect(subject.valid?).to be false
+      expect(subject.errors[:name]).to include("can't be blank")
+      expect(subject.errors[:institution]).to include("can't be blank")
+      expect(subject.errors[:device_model]).to include("can't be blank")
```

Comment:
How about adding `validate_presence_of` as a custom spec helper? It's more expressive to use that method than manually checking for error messages. #1491 adds more usages of that helper.

Reply:
I did that at first, but quickly removed it as there was so few occurrences. To be honest, I'd just remove those tests: they'll likely never break, and thus have zero value as tests.

Reply:
That's a good point. I've been thinking about that as well. When you have `validate_presence_of :name` in the model and `expect_it.to validate_presence_of :name` in the test, there's not much gained from having pretty much the same code twice. It still gives a safety net for unintentionally messing with the model, though.
It might not just prevent someone removing the validation from the model by mistake (this should be easy to spot in a review), but also side effects from other code changes. For example, if a pre-validation hook automatically assigns a value to `name` a failing validation test would alert that something might be odd.

Reply:
Sorry, I don't get your example :confused:

Reply:
I mean when you add something like `before_validation  { this.name = "Default Name" }`. This doesn't technically contradict presence validation but it would be odd. For example, the user would be allowed to leave the name field blank.
This is just an example of a side effect that may have an influence on presence validation. Of course this one is still quite obvious, but there could be more subtle side effects.
All I'm saying is even if the expectation doesn't add much value, it still can provide some safety and peace of mind.

Reply:
I had forgotten, but starting from ActiveRecord 5 the `belongs_to` associations are already required by default; we must specify `required: false` when it's optional. The only usage of `validate_presence_of` goes down to a single occurrence (`Device#name`).

I removed the specs.